eosdis-nasa/earthdata-pub-api
# Edpub 1163 set event type programmatically on submit

# Description

Updates step promotion to set event type programmatically.  This is a major shift in the workflow system and will require extensive testing.

## Spec

See Ticket: [EDPUB-1163](https://bugs.earthdata.nasa.gov/browse/EDPUB-1163)

---

## Validation

1. Make sure all merge request checks have passed (CI/CD).
2. Pull related branches locally.
3. Start the EDPub stack locally.
4. Navigate to localhost:3000.
5. Start a submission for all enabled DAACs.
6. Run all submission to publication.
7. Ensure all steps function as intended and EDPub internal notifications work as expected.
8. Deploy to sit.
9. Repeat 5-7 in sit.

***Emails will likely be broken this is not tested but is expected.  There is a ticket for next sprint to update the email system that will address this fully.


---

## Change Log

Updates step promotion to update set event type programmatically.

* [Fix _____](https://github.com/eosdis-nasa/earthdata-pub-api/commit/ef8e09685c999e4def0de6a5e789d01e0bb39c17)
